### PR TITLE
remove support for v2 UNSUPPORTED_REST_LEGACY

### PR DESCRIPTION
**Pull request by ankatare**
Signed-off-by: Abhay Narayan Katare <abhay.katare@india.nec.com>

Commit Message: remove support for v2 UNSUPPORTED_REST_LEGACY
Additional Description: Another PR of type remove_v2_support. first PR is https://github.com/envoyproxy/envoy/pull/16274
Risk Level: LOW
Testing: Yes
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by ankatare** [comment]
@htuch : please review

File: test/common/config/subscription_factory_impl_test.cc [hunk 1]
```diff
@@ -230,8 +230,6 @@ TEST_F(SubscriptionFactoryTest, FilesystemCollectionSubscriptionNonExistentFile)
 TEST_F(SubscriptionFactoryTest, LegacySubscription) {
   envoy::config::core::v3::ConfigSource config;
   auto* api_config_source = config.mutable_api_config_source();
-  api_config_source->set_api_type(
```
Comment (htuch) [inline test/common/config/subscription_factory_impl_test.cc:233]: Shouldn't the entire test just be deleted?

File: test/common/config/utility_test.cc [hunk 2]
```diff
@@ -74,9 +74,6 @@ TEST(UtilityTest, TranslateApiConfigSource) {
   Utility::translateApiConfigSource("test_rest_legacy_cluster", 10000,
                                     ApiType::get().UnsupportedRestLegacy,
                                     api_config_source_rest_legacy);
-  EXPECT_EQ(
```
Comment (htuch) [inline test/common/config/utility_test.cc:77]: Can you update this test to use some supported API type?

File: source/common/config/utility.cc [hunk 3]
```diff
@@ -44,10 +44,7 @@ void Utility::translateApiConfigSource(
     envoy::config::core::v3::GrpcService* grpc_service = api_config_source.add_grpc_services();
     grpc_service->mutable_envoy_grpc()->set_cluster_name(cluster);
   } else {
-    if (api_type == ApiType::get().UnsupportedRestLegacy) {
-      api_config_source.set_api_type(envoy::config::core::v3::ApiConfigSource::
-                                         hidden_envoy_deprecated_UNSUPPORTED_REST_LEGACY);
-    } else if (api_type == ApiType::get().Rest) {
+    if (api_type == ApiType::get().Rest) {
```
Comment (htuch) [inline source/common/config/utility.cc:47]: I think this should be just:
```
} else {
  ASSERT(api_type == ApiType::get().Rest);
  api_config_source.set_api_type(envoy::config::core::v3::ApiConfigSource::REST);
...
```
You should also remove `UnsupportedRestLegacy` and any references to this in the code in this PR.

**Review by htuch — CHANGES_REQUESTED** [review]
Thanks, a few comments.

**Comment by ankatare** [comment]
@htuch Thanks for review. i will get back ASAP

File: source/common/config/utility.cc (hunk 3 above)
Comment (ankatare) [reply source/common/config/utility.cc:47]: Done. Thnaks for review

File: test/common/config/subscription_factory_impl_test.cc (hunk 1 above)
Comment (ankatare) [reply test/common/config/subscription_factory_impl_test.cc:233]: Yeah.. done. Thnaks for review

**Comment by htuch** [comment]
@ankatare you need to push again

**Comment by ankatare** [comment]
@htuch yeah... actually build failing locally .. will push soon

File: test/common/config/utility_test.cc (hunk 2 above)
Comment (ankatare) [reply test/common/config/utility_test.cc:77]: Done. Sorry for delay :(

**Comment by ankatare** [comment]
@phlax please review

File: test/common/config/subscription_factory_impl_test.cc [hunk 4]
```diff
@@ -220,18 +220,9 @@ TEST_F(SubscriptionFactoryTest, FilesystemCollectionSubscription) {
   collectionSubscriptionFromUrl(fmt::format("file:///{}", file_path), {})->start({});
 }
 
-TEST_F(SubscriptionFactoryTest, FilesystemCollectionSubscriptionNonExistentFile) {
```
Comment (htuch) [inline test/common/config/subscription_factory_impl_test.cc:223]: How is this test relevant?

**Review by htuch — CHANGES_REQUESTED** [review]
Looks good, just one unrelated change I think.
/wait

File: test/common/config/subscription_factory_impl_test.cc (hunk 4 above)
Comment (ankatare) [reply test/common/config/subscription_factory_impl_test.cc]: @htuch  Really Sorry, My mistake :( . reverted back it and deleted  Test Cae "LegacySubscription"

**Review by htuch — APPROVED** [review]
LGTM, thanks!
